internal/factory/daemonset.go

```diff
@@ -8,11 +8,12 @@ import (
 )
 
 // NewDaemonSet stubs an instance of a daemonset
-func NewDaemonSet(daemonsetName, namespace, loggingComponent, component string, podSpec core.PodSpec) *apps.DaemonSet {
+func NewDaemonSet(daemonsetName, namespace, loggingComponent, component, impl string, podSpec core.PodSpec) *apps.DaemonSet {
 	labels := map[string]string{
-		"provider":      "openshift",
-		"component":     component,
-		"logging-infra": loggingComponent,
+		"provider":       "openshift",
+		"component":      component,
+		"logging-infra":  loggingComponent,
+		"implementation": impl,
```

Comment:
This can not be part of the label-set used as the `Selector`, as that set can not be updated while the DaemonSet exists, making a change of implementation without completely deleting the DaemonSet impossible.

A possible solution would be to use the previous label-set as the selector and extend this with the new `implementation` label only for the `ObjectMeta.Labels` values.